aws-datastore/src/main/java/com/amplifyframework/datastore/storage/LocalStorageAdapter.java

```diff
@@ -75,6 +75,22 @@ void initialize(
             @NonNull StorageItemChange.Initiator initiator,
             @NonNull ResultListener<StorageItemChange.Record> itemSaveListener);
 
+    /**
+     * Save an item into local storage only if the data being overwritten meets the
```

Comment:
`overwritten` --> `written`. 

On this API, `@Nullable` predicate is okay, since this is an internal interface.

Reply:
just emphasizing that predicate is being applied to data already inside storage, not the new data being written :)